src/global_sequences_viewer.cpp

```diff
@@ -71,15 +71,10 @@ Status GlobalSequencesViewer::removeIdleSequences() {
     return ovms::StatusCode::OK;
 }
 
-std::cv_status GlobalSequencesViewer::waitFor(uint32_t sequenceCleanerInterval) {
-    std::unique_lock<std::mutex> lock(cleanerControlMutex);
-    return cleanerControlCv.wait_for(lock, std::chrono::minutes(sequenceCleanerInterval));
-}
-
-void GlobalSequencesViewer::sequenceCleanerRoutine(uint32_t sequenceCleanerInterval) {
+void GlobalSequencesViewer::sequenceCleanerRoutine(uint32_t sequenceCleanerInterval, std::future<void> exitSignal) {
     SPDLOG_LOGGER_INFO(modelmanager_logger, "Started sequence cleaner thread");
 
-    while (waitFor(sequenceCleanerInterval) == std::cv_status::timeout) {
+    while (exitSignal.wait_for(std::chrono::minutes(sequenceCleanerInterval)) == std::future_status::timeout) {
```

Comment:
Minor:
Usually when dealing with units of time in code it is usefull to mention in name its unit type -> in this case
sequenceCleanerInterval -> sequenceCleanerIntervalMinutes